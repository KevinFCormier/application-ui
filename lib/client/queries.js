import gql from 'graphql-tag'

export const HCMClusterList = gql`
    query getClusters {
        items: clusters {
            ClusterName
            Status
            TotalNodes
            TotalDeployments
        }
    }
`

export const HCMPodList = gql`
    query getPods {
        items: pods {
            Namespace
            PodName
            name
            cluster
        }
    }
`
export const HCMNodeList = gql`
    query getNodes {
        items: nodes {
            NodeName
            name
            cluster
            NodeDetails {
                Status
                OSImage
                Cpu
            }
        }
    }
`

export const HCMPersistentVolumeList = gql`
    query getPVS {
        items: pvs {
            PVName
            name
            cluster
            PVDetails {
                Status
                StorageClass
                Capacity
            }
        }
    }
`
