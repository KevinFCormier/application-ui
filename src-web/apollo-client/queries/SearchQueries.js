/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import gql from 'graphql-tag'

export const GET_SEARCH_COMPLETE = gql`
  query searchComplete($property: String!, $query: SearchInput) {
    searchComplete(property: $property, query: $query)
  }
`

export const GET_SEARCH_INPUT_TEXT = gql`
  {
    searchInput @client {
      text
    }
  }
`

export const GET_SEARCH_TABS = gql`
  {
    searchQueryTabs @client {
      unsavedCount
      openedTabName
      tabs {
        queryName
        searchText
        description
        updated
        id
      }
    }
  }
`

export const SEARCH_QUERY = gql`
  query searchResult($input: [SearchInput]) {
    searchResult: search(input: $input){
      count
      items
      updatedTimestamp
    }
  }
`

export const GET_SAVED_USER_QUERY = gql`
  query userQueries {
    items: userQueries {
      name
      description
      searchText
    }
  }
`

export const SAVE_USER_QUERY = gql`
  mutation saveQuery($resource: JSON!) {
    saveQuery(resource: $resource)
  }
`
