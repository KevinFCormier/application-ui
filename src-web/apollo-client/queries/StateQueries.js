/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import gql from 'graphql-tag'

export const GET_MODAL_STATE = gql`
  {
    modal @client {
      open
      type
      data {
        name
        description
        searchText
      }
    }
  }
`

export const GET_ACTION_MODAL_STATE = gql`
  {
    actionModal @client {
      open
      type
      resourceType {
        name
        list
      }
      data {
        item
      }
    }
  }
`

export const UPDATE_MODAL = gql`
  mutation UpdateModal($__typename: String, $open: Boolean, $type: String, $data: JSON) {
    updateModal(__typename: $__typename, open:$open, type:$type, data:$data) @client
  }
`

export const UPDATE_ACTION_MODAL = gql`
  mutation UpdateActionModal($__typename: String, $open: Boolean, $type: String, $resourceType: JSON, $data: JSON) {
    updateActionModal(__typename: $__typename, open:$open, type:$type, resourceType:$resourceType, data:$data) @client
  }
`

export const UPDATE_QUERY_TABS = gql`
  mutation UpdateQueryTabs($__typename: String, $unsavedCount: Int, $openedTabName: String, $openedTabId: String, $data: JSON, $tabs: [JSON]) {
    updateQueryTabs(__typename: $__typename, unsavedCount:$unsavedCount, openedTabName:$openedTabName, openedTabId:$openedTabId, data:$data, tabs: $tabs) @client
  }
`

export const UPDATE_SINGLE_QUERY_TAB = gql`
  mutation UpdateSingleQueryTab($openedTabName: String, $openedTabId: String, $description: String, $searchText: String, $updateUnsavedOrExisting: Boolean) {
    updateSingleQueryTab(openedTabName: $openedTabName, openedTabId: $openedTabId, description: $description, searchText:$searchText, updateUnsavedOrExisting:$updateUnsavedOrExisting) @client
  }
`

export const REMOVE_SINGLE_QUERY_TAB = gql`
  mutation RemoveSingleQueryTab($id: String) {
    removeSingleQueryTab(id: $id) @client
  }
`
