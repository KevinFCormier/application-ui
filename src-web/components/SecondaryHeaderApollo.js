/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import { Tabs, Icon } from 'carbon-components-react'
import Tab from './common/Tab'
import { DetailPageHeader } from 'carbon-addons-cloud-react'
import resources from '../../lib/shared/resources'
import msgs from '../../nls/platform.properties'
import { Query } from 'react-apollo'
import { UPDATE_QUERY_TABS, REMOVE_SINGLE_QUERY_TAB } from '../apollo-client/queries/StateQueries'
import { GET_SEARCH_TABS } from '../apollo-client/queries/SearchQueries'
import _ from 'lodash'

resources(() => {
  require('../../scss/secondary-header.scss')
})

export class SecondaryHeaderSearchPage extends React.Component {
  constructor(props) {
    super(props)
    this.renderTabs = this.renderTabs.bind(this)
    this.state = {
      //eslint-disable-next-line
      isHovering: false,
    }
  }

  handleMouseHover = () => {
    this.setState((preState) => {return { isHovering: !preState.isHovering }})
  }


  render() {
    const { title } = this.props
    const { locale } = this.context
    return (
      <Query query={GET_SEARCH_TABS}>
        {( { data, client } ) => {
          const tabs = _.get(data, 'searchQueryTabs.tabs')
          const unsavedCount = _.get(data, 'searchQueryTabs.unsavedCount')
          const openedTabName = _.get(data, 'searchQueryTabs.openedTabName')
          return (
            <div className='secondary-header-wrapper' role='region' aria-label={title}>
              <div className='secondary-header'>
                {tabs && tabs.length > 0 ? (
                  <DetailPageHeader hasTabs={true} title={decodeURIComponent(title)} aria-label={`${title} ${msgs.get('secondaryHeader', locale)}`}>
                    <Tabs className='secondary-header-tabs' selected={this.getSelectedTab(tabs, openedTabName) || 0} aria-label={`${title} ${msgs.get('tabs.label', locale)}`}>
                      {this.renderTabs(client, tabs, unsavedCount)}
                      {this.renderAddNewTab(client, unsavedCount, tabs, locale)}
                    </Tabs>
                  </DetailPageHeader>
                ) : (
                  <DetailPageHeader hasTabs={true} title={decodeURIComponent(title)} aria-label={`${title} ${msgs.get('secondaryHeader', locale)}`}>
                  </DetailPageHeader>
                )}
              </div>
            </div>
          )
        }}
      </Query>
    )
  }

  renderAddNewTab = (client, unsavedCount, tabs, locale) => {
    return (
      <Tab label={'+' + msgs.get('tabs.add.new', locale)}
        id={'add-new-query'} onClick={this.handleClickNewTab(client, unsavedCount, tabs)} />
    )
  }

  renderTabs(client, tabs, unsavedCount) {
    return tabs.map((tab) => {
      return (
        <Tab label={`${tab.queryName} ${tab.updated ? '*': ''}`}
          key={tab.queryName}
          id={tab.queryName}
          onClick={(evt) => {
            if (evt.target.nodeName === 'A' || evt.target.nodeName === 'LI') this.handleClickTab(client, tabs, tab, unsavedCount)}
          }
          onMouseEnter={this.handleMouseHover}
          onMouseLeave={this.handleMouseHover}
          subcomponent = { tabs.length > 1 ?
            <Icon
              className='header-icon--close'
              name='icon--close'
              fill={'#5a6872'}
              description={msgs.get('tabs.close.icon', this.context.locale)}
              onClick={this.handleRemoveClick(client, tab)} /> : null
          }
        />
      )
    })
  }

  handleRemoveClick = (client, tab) => async () => {
    const { queryName } = tab
    const newData =  {
      queryName
    }
    await client.mutate({ mutation: REMOVE_SINGLE_QUERY_TAB, variables: { ...newData } })
  }

  handleClickNewTab = (client, unsavedCount, tabs) => async () => {
    const newData =  {
      __typename: 'SearchQueryTabs',
      unsavedCount: unsavedCount + 1,
      openedTabName: `Unsaved-${unsavedCount}`,
      data:{
        queryName: `Unsaved-${unsavedCount}`,
        searchText:'',
        description:'',
        id: `Unsaved-${unsavedCount}`,
        updated: false,
        __typename: 'QueryTab'
      },
      tabs
    }
    await client.mutate({ mutation: UPDATE_QUERY_TABS, variables: { ...newData } })

    client.writeData({ data: {
      searchInput: {
        __typename: 'SearchInput',
        text: ''
      }
    }} )
  }

  async handleClickTab(client, tabs, tab, unsavedCount) {
    const { queryName, searchText } = tab
    const newData =  {
      __typename: 'SearchQueryTabs',
      unsavedCount: unsavedCount,
      openedTabName: queryName,
      tabs
    }
    await client.mutate({ mutation: UPDATE_QUERY_TABS, variables: { ...newData } })

    client.writeData({ data: {
      searchInput: {
        __typename: 'SearchInput',
        text: searchText
      }
    }} )
  }

  getSelectedTab(tabs, openedTabName) {
    const index = tabs.findIndex((tab) =>
      tab.queryName === openedTabName
    )
    return index === -1 ? 0 : index
  }

}

SecondaryHeaderSearchPage.contextTypes = {
  locale: PropTypes.string
}

SecondaryHeaderSearchPage.propTypes = {
  title: PropTypes.string
}

export default SecondaryHeaderSearchPage
