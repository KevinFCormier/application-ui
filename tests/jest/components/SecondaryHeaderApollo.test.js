/*******************************************************************************
 * Licensed Materials - Property of IBM
 * 5737-E67
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
'use strict'
/* eslint-disable import/no-named-as-default */

import React from 'react'
import { shallow } from 'enzyme'
import { MockedProvider } from 'react-apollo/test-utils'

import SecondaryHeaderApollo from '../../../src-web/components/SecondaryHeaderApollo'
import { GET_SEARCH_TABS } from '../../../src-web/apollo-client/queries/SearchQueries'

const mocks = {
  getSearchTabsMock: {
    request: { query: GET_SEARCH_TABS },
    result: {
      data: {
        searchQueryTabs: {
          tabs: [
            {
              queryName: 'UnitTesting',
              searchText: 'kind:pod',
              description: 'Search for pods',
              updated: false,
              id: 'UnitTesting'
            },
            {
              queryName: 'UnitTesting-1',
              searchText: 'kind:pod',
              description: 'Search for pods 1',
              updated: false,
              id: 'UnitTesting-1'
            }
          ],
          unsavedCount: 2,
          openedTabName: 'UnitTesting',
          openedTabId: 'UnitTesting'
        }
      }
    }
  }
}

const delay = ms =>
  new Promise(resolve => {
    setTimeout(() => {
      resolve()
    }, ms)
  })

describe('SecondaryHeaderApollo component 1', () => {
  it('Changes Apollo Client Cache For Edit Modal', async () => {
    const component = shallow(
      <MockedProvider mocks={[mocks.getSearchTabsMock]} addTypename={false}>
        <SecondaryHeaderApollo title="hello world" />
      </MockedProvider>
    )
    await delay(0)
    expect(component.find(SecondaryHeaderApollo)).toMatchSnapshot()
  })
})
