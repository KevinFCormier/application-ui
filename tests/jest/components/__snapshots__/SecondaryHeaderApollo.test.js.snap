// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryHeaderApollo component 1 Changes Apollo Client Cache For Edit Modal 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <MockedProvider
      addTypename={false}
      mocks={
        Array [
          Object {
            "request": Object {
              "query": Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": undefined,
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [
                            Object {
                              "arguments": Array [],
                              "kind": "Directive",
                              "name": Object {
                                "kind": "Name",
                                "value": "client",
                              },
                            },
                          ],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "searchQueryTabs",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "unsavedCount",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "openedTabName",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "openedTabId",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "tabs",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "queryName",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "searchText",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "description",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "updated",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "id",
                                      },
                                      "selectionSet": undefined,
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 207,
                  "start": 0,
                },
              },
            },
            "result": Object {
              "data": Object {
                "searchQueryTabs": Object {
                  "openedTabId": "UnitTesting",
                  "openedTabName": "UnitTesting",
                  "tabs": Array [
                    Object {
                      "description": "Search for pods",
                      "id": "UnitTesting",
                      "queryName": "UnitTesting",
                      "searchText": "kind:pod",
                      "updated": false,
                    },
                    Object {
                      "description": "Search for pods 1",
                      "id": "UnitTesting-1",
                      "queryName": "UnitTesting-1",
                      "searchText": "kind:pod",
                      "updated": false,
                    },
                  ],
                  "unsavedCount": 2,
                },
              },
            },
          },
        ]
      }
    >
      <SecondaryHeaderApollo
        title="hello world"
      />
    </MockedProvider>,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <SecondaryHeaderApollo
          title="hello world"
        />,
        "client": ApolloClient {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "defaultOptions": Object {},
          "disableNetworkFetches": false,
          "link": ApolloLink {
            "request": [Function],
          },
          "mutate": [Function],
          "query": [Function],
          "queryDeduplication": true,
          "reFetchObservableQueries": [Function],
          "resetStore": [Function],
          "resetStoreCallbacks": Array [],
          "ssrMode": false,
          "store": DataStore {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
          },
          "version": "2.4.5",
          "watchQuery": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": "hello world",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <SecondaryHeaderApollo
            title="hello world"
          />,
          "client": ApolloClient {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
            "defaultOptions": Object {},
            "disableNetworkFetches": false,
            "link": ApolloLink {
              "request": [Function],
            },
            "mutate": [Function],
            "query": [Function],
            "queryDeduplication": true,
            "reFetchObservableQueries": [Function],
            "resetStore": [Function],
            "resetStoreCallbacks": Array [],
            "ssrMode": false,
            "store": DataStore {
              "cache": InMemoryCache {
                "addTypename": false,
                "cacheKeyRoot": KeyTrie {
                  "weakness": true,
                },
                "config": Object {
                  "addTypename": false,
                  "dataIdFromObject": [Function],
                  "fragmentMatcher": HeuristicFragmentMatcher {},
                  "freezeResults": false,
                  "resultCaching": true,
                },
                "data": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "maybeBroadcastWatch": [Function],
                "optimisticData": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "silenceBroadcast": false,
                "storeReader": StoreReader {
                  "executeSelectionSet": [Function],
                  "executeStoreQuery": [Function],
                  "executeSubSelectedArray": [Function],
                  "freezeResults": false,
                },
                "storeWriter": StoreWriter {},
                "typenameDocumentCache": Map {},
                "watches": Set {},
              },
            },
            "version": "2.4.5",
            "watchQuery": [Function],
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": "hello world",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "title": "hello world",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "title": "hello world",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <SecondaryHeaderApollo
          title="hello world"
        />,
        "client": ApolloClient {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "defaultOptions": Object {},
          "disableNetworkFetches": false,
          "link": ApolloLink {
            "request": [Function],
          },
          "mutate": [Function],
          "query": [Function],
          "queryDeduplication": true,
          "reFetchObservableQueries": [Function],
          "resetStore": [Function],
          "resetStoreCallbacks": Array [],
          "ssrMode": false,
          "store": DataStore {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
          },
          "version": "2.4.5",
          "watchQuery": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": "hello world",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
}
`;
